sdk/resourcemanager/Azure.ResourceManager/src/Custom/Resources/Models/UserAssignedIdentity.cs

```diff
@@ -4,20 +4,31 @@
 using System;
 using System.ComponentModel;
 using Azure.Core;
+using Azure.ResourceManager.Core;
 
 namespace Azure.ResourceManager.Resources.Models
 {
     /// <summary>
     /// A class representing an Identity assigned by the user.
     /// </summary>
+    [PropertyReferenceType]
     public sealed partial class UserAssignedIdentity : IEquatable<UserAssignedIdentity>
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UserAssignedIdentity"/> class.
+        /// </summary>
+        [InitializationConstructor]
+        public UserAssignedIdentity()
+        {
+        }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="UserAssignedIdentity"/> class.
         /// </summary>
         /// <param name="clientId"> ClientId . </param>
         /// <param name="principalId"> PrincipalId. </param>
-        public UserAssignedIdentity(Guid clientId, Guid principalId)
+        [SerializationConstructor]
+        internal UserAssignedIdentity(Guid clientId, Guid principalId)
```

Comment:
I know Michael had test case stipulate SerializationCtor should be internal. I had liked the public ctor where forces you to pass in clientid and principalId instead of potentially optionally/partially initialized object.

Reply:
Actually, an empty UserAssignedIdentity is needed in the request when creating a resource with a user assigned identity. It's usually defined as `IDictionary<string, UserAssignedIdentity> UserAssignedIdentities` and you need to pass the key with an empty object in resource creation. An [example request ](https://github.com/Azure/azure-rest-api-specs/blob/88e7838a09868a51de3894114355c75929847a46/specification/mediaservices/resource-manager/Microsoft.Media/stable/2021-06-01/examples/accounts-create.json#L15-L18) will have:
```
"userAssignedIdentities": {
      "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id1": {},
      "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/id2": {}
}
```